cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1223,35 +1223,26 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
         Future<ReconciliationState> getZookeeperDescription() {
             Promise<ReconciliationState> promise = Promise.promise();
 
-            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
-                future -> {
-                    try {
-                        StatefulSet sts = zkSetOperations.get(namespace, ZookeeperCluster.zookeeperClusterName(name));
-                        Storage oldStorage = getOldStorage(sts);
+            zkSetOperations.getAsync(namespace, ZookeeperCluster.zookeeperClusterName(name)).setHandler(res -> {
```

Comment:
I wonder if we can further simplify this by using the `compose` method. It could look something like this:

```java
return zkSetOperations.getAsync(namespace, ZookeeperCluster.zookeeperClusterName(name))
                    .compose(sts -> {
                        Storage oldStorage = getOldStorage(sts);

                        this.zkCluster = ZookeeperCluster.fromCrd(kafkaAssembly, versions, oldStorage);
                        this.zkService = zkCluster.generateService();
                        this.zkHeadlessService = zkCluster.generateHeadlessService();

                        if (zkCluster.getLogging() instanceof ExternalLogging)  {
                            return configMapOperations.getAsync(kafkaAssembly.getMetadata().getNamespace(), ((ExternalLogging) zkCluster.getLogging()).getName());
                        } else {
                            return Future.succeededFuture(null);
                        }
                    })
                    .compose(cm -> {
                        ConfigMap logAndMetricsConfigMap = zkCluster.generateMetricsAndLogConfigMap(cm);
                        this.zkMetricsAndLogsConfigMap = zkCluster.generateMetricsAndLogConfigMap(logAndMetricsConfigMap);

                        return Future.succeededFuture(this);
                    });
```

I think the same can be applied to all of the other methods you changed.